### PR TITLE
Add reference docs for CLI

### PR DESCRIPTION
**Pull request by Dinoosawruss**
Issue: #8 

I have tried to reference all the current CLI and build a format in line with my preferences from parts of the [AWS CLI](https://docs.aws.amazon.com/cli/latest/reference), [Prisma CLI](https://www.prisma.io/docs/reference/api-reference/command-reference) and [Stripe CLI](https://stripe.com/docs/cli) reference guides. 

Hopefully, it covers everything and is in a style that is helpful and easy to follow

If you would like any changes made or if I have missed anything do let me know

### CODE REVIEW

File: docs/cli-reference.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# StepCI CLI reference
```
Comment (mishushakov) [inline docs/cli-reference.md:1]: Use "CLI Reference" instead of "StepCI CLI reference"

File: docs/cli-reference.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+The following documents evety command and flag avaliable for use in StepCI's command-line interface.
```
Comment (mishushakov) [inline docs/cli-reference.md:2]: Fix grammar: "evety" and "avaliable". Also Step CI is written with a space

File: docs/cli-reference.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+With the CLI you can:
+- Run workflows
+
```
Comment (mishushakov) [inline docs/cli-reference.md:6]: This is unnecessary and can be removed

File: docs/cli-reference.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+Follow the [getting started guide](https://github.com/stepci/stepci/blob/main/docs/get-started.md) for setup instructions
```
Comment (mishushakov) [inline docs/cli-reference.md:7]: Change the link to (../get-started.md). Also change setup instructions > installation instructions

File: docs/cli-reference.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+## Synopsis
+---
+The `stepci` command can be called once installed and is the main command for the CLI.
+
+The primary method to execute StepCI workflows is `stepci run [command]`
```
Comment (mishushakov) [inline docs/cli-reference.md:13]: execute StepCI workflows > execute workflows

File: docs/cli-reference.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+Options:
+  --help     Show help                                                 [boolean]
+  --version  Show version number                                       [boolean]
+  ```
+
+## Commands
+---
+### `run [workflow]`
+The `run ` command allows you to execute a specified workflow
```
Comment (mishushakov) [inline docs/cli-reference.md:31]: allows you to execute > lets you execute

File: docs/cli-reference.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+#### **Options**
+| Options | Required | Description |
+|-|-|-|
+| --help | No | Outputs the help text for `stepci run` |
+| --version | No | Outputs the current stepci version |
+<br>
+
+#### **Examples**
+Run test workflow to return the status of [example.com](https://example.com)
```
Comment (mishushakov) [inline docs/cli-reference.md:47]: Run test workflow to return the status > Run example workflow that checks the status

File: docs/cli-reference.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+✔ 200
+
+
+✔ Status Check (./examples/status.yml) passed in 0.526s
+```
+
+## Options
+---
+### `--help`
+Outputs the help menu for the main `stepci` comamnd
```
Comment (mishushakov) [inline docs/cli-reference.md:68]: Fix grammar: "comamnd"

File: docs/cli-reference.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+Commands:
+  run [workflow]  run workflow
+
+Options:
+  --help     Show help                                                 [boolean]
+  --version  Show version number                                       [boolean]
+```
+
+### `--version`
+Outputs the current version of stepci installed
```
Comment (mishushakov) [inline docs/cli-reference.md:84]: Outputs the current version of stepci installed > Outputs the version currently installed

File: docs/cli-reference.md [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+Follow the [getting started guide](https://github.com/stepci/stepci/blob/main/docs/get-started.md) for setup instructions
+
+## Synopsis
+---
+The `stepci` command can be called once installed and is the main command for the CLI.
+
+The primary method to execute StepCI workflows is `stepci run [command]`
+
+When `stepci` is called with the `--help` option a list of avaliable commands and options will be displayed:
```
Comment (mishushakov) [inline docs/cli-reference.md:15]: add a comma after "option"

File: docs/cli-reference.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+## Commands
+---
+### `run [workflow]`
+The `run ` command allows you to execute a specified workflow
+
+#### **Arguments**
+| Argument | Required | Description |
+|-|-|-|
+| workflow | Yes | Specify the path to a stepci workflow |
```
Comment (mishushakov) [inline docs/cli-reference.md:36]: stepci workflow > workflow

File: docs/cli-reference.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+<br>
+
+#### **Options**
+| Options | Required | Description |
+|-|-|-|
+| --help | No | Outputs the help text for `stepci run` |
+| --version | No | Outputs the current stepci version |
```
Comment (mishushakov) [inline docs/cli-reference.md:43]: current stepci version > current version

**Review by mishushakov — CHANGES_REQUESTED** [review]
Fantastic work! Only small changes needed before we can merge. Thank you so much so far!

**Comment by mishushakov** [comment]
Also don't forget to change `docs/README.md` to include `[CLI Reference](cli-reference.md)`

**Comment by Dinoosawruss** [comment]
Should all be resolved

File: docs/cli-reference.md [hunk 13]
```diff
@@ -28,12 +25,12 @@ Options:
 ## Commands
 ---
 ### `run [workflow]`
-The `run ` command allows you to execute a specified workflow
+The `run ` command lets you execute a specified workflow
```
Comment (mishushakov) [inline docs/cli-reference.md:28]: Remove space after run

File: docs/README.md [hunk 14]
```diff
@@ -3,3 +3,4 @@
 - [Get Started](./get-started.md)
 - [Using in test tools](./using-test-tools.md)
 - [Workflow Syntax](./workflow-syntax.md)
+- [CLI Reference](./cli-reference.md)
```
Comment (mishushakov) [inline docs/README.md:6]: I think we need to reorder this bit

1. Get Started
2. CLI Reference
3. Workflow Syntax
4. Using in Test Tools

**Review by mishushakov — APPROVED** [review]
Amazing work. We're merging it!
Thank you Michael so much!

**Comment by mishushakov** [comment]
By the way, I've adjusted your CLI reference a little bit: https://github.com/stepci/stepci/commit/dd0d1c2e6e4267efc4ddfb50536e5ca8e760ea6d

I learned that I shouldn't be too picky on pull requests, because making some small changes myself takes less time than actually communicating that these changes should to be made 🤷

**Comment by sebastianwis** [comment]
Thank you so much, @Dinoosawruss !
We are happy to have you as contributor.
It would be amazing if you would like to accompany us on the further journey in any way!

**Pull request merged**